plugins/obs-ffmpeg/texture-amf.cpp

```diff
@@ -1385,10 +1424,69 @@ try {
 	return false;
 }
 
+static inline void check_recommended_avc_defaults(amf_base *enc,
+						  obs_data_t *settings)
+{
+	int64_t framerate = enc->fps_num / enc->fps_den;
+	if ((enc->cx * enc->cy > 1920 * 1088) || (framerate > 60)) {
+
+		// Recommended base defaults
+		obs_data_set_default_int(settings, "bitrate", 2500);
+		obs_data_set_default_int(settings, "cqp", 20);
+		obs_data_set_default_string(settings, "rate_control", "CBR");
+		obs_data_set_default_string(settings, "preset", "quality");
+		obs_data_set_default_string(settings, "profile", "high");
+		obs_data_set_default_int(settings, "bf", 0);
+		info("Original default settings were lowered according to resolution"
+		     " and framerate.");
+	} else {
+		// Recommended high perceptual quality defaults
+		// for up to 1080p60fps
+		set_avc_property(enc, ADAPTIVE_MINIGOP, true);
+
+		bool pa_enabled = obs_data_get_bool(settings, "pre_analysis");
+		bool pa_set_by_user =
+			obs_data_has_user_value(settings, "pre_analysis");
+
+		if (!pa_set_by_user || (pa_set_by_user && pa_enabled)) {
+			pa_enabled = true;
+			obs_data_set_bool(settings, "pre_analysis", pa_enabled);
+
+			set_avc_property(enc, PRE_ANALYSIS_ENABLE, pa_enabled);
+			set_amf_property(enc, AMF_PA_LOOKAHEAD_BUFFER_DEPTH,
+					 20);
+			set_amf_property(enc, AMF_PA_TAQ_MODE,
+					 AMF_PA_TAQ_MODE_2);
+			set_amf_property(enc, AMF_PA_ENGINE_TYPE,
+					 AMF_MEMORY_OPENCL);
+
+			info("High perceptual quality defaults:\n"
+			     "\tAdaptiveMiniGOP:        %s\n"
+			     "\tEnablePreAnalysis:      %s\n"
+			     "\tPALookAheadBufferDepth: %d\n"
+			     "\tPATemporalAQMode:       %d\n"
+			     "\tPAEngineType:           %s\n",
+			     "true", "true", 20, AMF_PA_TAQ_MODE_2, "OpenCL");
```

Comment:
I'm not sure if I'd prefer a short string for `PATemporalAQMode`, such as "MODE_NONE", "MODE_1", or "MODE_2", instead of the raw enum value. I'll have to think about it, but I wanted to comment so I didn't forget.

Reply:
This is probably fine as-is for now.